### PR TITLE
[Py3][ENG-129] Even more Py3 backwards compatibility changes

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Get code in line before we switch over to Python 3.

## Changes

- Removed lambda parenthesis 
- change relative imports to absolutes
- moved urllib library to future
- changed the http status library to the DRF one which is good in both 2 and 3

## QA Notes

No user facing changes
Check links of addon urls to see if there are any changes in logging and display. Check displayed links and connect/disconnect from Gitlab, Bitbucket and Github.

## Documentation

Tech task, no docs

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-150
https://openscience.atlassian.net/browse/ENG-129

### CODE REVIEW

File: addons/wiki/utils.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def broadcast_to_sharejs(action, sharejs_uuid, node=None, wiki_name='home', data
     )
 
     if action == 'redirect' or action == 'delete':
-        redirect_url = urllib.quote(
+        redirect_url = quote(
```
Comment (brianjgeiger) [inline addons/wiki/utils.py]: Could be tested

File: addons/zotero/tests/test_views.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 import mock
 import pytest
-import urlparse
+from future.moves.urllib.parse import urlparse, urljoin
```
Comment (UdayVarkhedkar) [inline addons/zotero/tests/test_views.py]: Is importing `urlparse` necessary here?

File: addons/twofactor/tests/test_models.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import unittest
-from urlparse import parse_qs, urlparse
+from future.moves.urllib.parse import urlparse, urljoin, parse_qs
```
Comment (UdayVarkhedkar) [inline addons/twofactor/tests/test_models.py]: Is `urljoin` necessary here?

File: addons/bitbucket/utils.py [hunk 4]
```diff
@@ -9,9 +9,9 @@
 def get_path(kwargs, required=True):
     path = kwargs.get('path')
     if path:
-        return urllib.unquote_plus(path)
+        return unquote_plus(path)
```
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply addons/bitbucket/utils.py]: I see the importance/necessity  of having tests for these functions, because even a minor deviations could cause problems, but I don't think it's really efficient here because:

1. `urllib.unquote_plus` and `future.moves.urllib.parse.unquote_plus` are exactly not too versions of the same modules, they are exactly the same module, just moved to a place that's accessible to both 2 and 3

2. Even if I wrote tests for these functions (and they do need tests.) I wounldn't know what edge cases could potentially cause failures, I've tried stuff like `urlparse.urlparse('>+%%%%$%##%35%25<') == future_urlparse('>+%%%%$%##%35%25<')` and it works.

I do support raising coverage here, but in the interest of expedience we should put this off until later.

File: website/project/decorators.py [hunk 5]
```diff
@@ -42,10 +42,11 @@ def _kwargs_to_nodes(kwargs):
         if not node:
             node = _load_node_or_fail(pid)
     elif nid and not pid:
+
```
Comment (UdayVarkhedkar) [inline website/project/decorators.py]: Is this blank line intentional?

File: osf/utils/tokens.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
+from rest_framework import status as http_status
```
Comment (UdayVarkhedkar) [inline osf/utils/tokens.py]: It looks like this file was removed in this commit (https://github.com/CenterForOpenScience/osf.io/commit/d4f5586a719e37c5c2b3a107008e3e986f98678b) because it duplicates code. I'm guessing it shouldn't be re-added and is a byproduct of branching/merging?

File: website/project/decorators.py [hunk 7]
```diff
@@ -438,20 +439,6 @@ def wrapped(*args, **kwargs):
     # Return decorated function
     return wrapped
 
-def http_error_if_disk_saving_mode(func):
```
Comment (UdayVarkhedkar) [inline website/project/decorators.py]: To confirm, this function was removed because it isn't in use anywhere (based on a grep), is that correct?

File: osf/utils/tokens.py (hunk 6 above)
Comment (Johnetordoff) [reply osf/utils/tokens.py]: It looks like it is being used in that file though. Ln40, Ln80

Comment (UdayVarkhedkar) [reply osf/utils/tokens.py]: Sorry, I'm not talking about the import. I was trying to say that `osf.utils.tokens.py` was removed in https://github.com/CenterForOpenScience/osf.io/commit/d4f5586a719e37c5c2b3a107008e3e986f98678b because it duplicates code.

Comment (Johnetordoff) [reply osf/utils/tokens.py]: good catch

File: addons/bitbucket/utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Johnetordoff) [reply addons/bitbucket/utils.py]: I don't usually write new tests when I don't add or modify new functionality. There also the issue in this particular PR that many of these areas are flask related and should be just be removed, but that's outside the scope.

I'm also not crazy about testing external libraries at all, they have there own tests. I feel it's better to just be deliberate when updating, in this case particalarly these libraries are well-known and not being used in complex ways.

**Comment by pattisdr** [comment]
LGTM, thanks John 🚢
